vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher.go

```diff
@@ -309,6 +312,43 @@ func (f *controllerFetcher) getOwnerForScaleResource(groupKind schema.GroupKind,
 	return nil, lastError
 }
 
+func (f *controllerFetcher) GetPodTemplateFromTopMostWellKnown(key *ControllerKeyWithAPIVersion) (*corev1.PodTemplateSpec, error) {
+	kind := wellKnownController(key.Kind)
+	informer, exists := f.informersMap[kind]
+	if !exists {
+		return nil, fmt.Errorf("can't retrieve podTemplate, no informer defined for kind %v", kind)
+	}
+
+	namespace := key.Namespace
+	name := key.Name
+
+	obj, exists, err := informer.GetStore().GetByKey(namespace + "/" + name)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("%s %s/%s does not exist", kind, namespace, name)
+	}
+	switch apiObj := obj.(type) {
```

Comment:
this part could be in its own function as well

Reply:
Agree, but I am trying to be consistent with the style, like in `getParentOfWellKnownController`